### PR TITLE
Sync: Add user password syncing as an event

### PR DESCRIPTION
**Pull request by enejb**
Add state of the jetpack_sync_save_user event so that we know a bit more information as to why it was triggered in the first place. This will make it easier to provide info to the user as to what was changed.

- Did the role change? Did the capabilities change. Did we also change the password? Did all those things change at once? 

Fixes issue where we didn't record that we a password was changed. 

#### Changes proposed in this Pull Request:
* Add a state to the jetpack_save_user sync event that provides more info as to why the event happened. 

#### Testing instructions:
* Do the tests pass? 
* Does everything still work as before? 
* Test changing password, Role, and user data at once. 
* Test adding a new user. Do we get the right info? 
* Test Removing a user? 
* Does multisite still work as expected?

### CODE REVIEW

File: sync/class.jetpack-sync-module-users.php [hunk 1]
```diff
@@ -203,8 +207,14 @@ function save_user_handler( $user_id, $old_user_data = null ) {
 		}
 
 		if ( $old_user !== null ) {
+			if ( $raw_user->user_pass !== $old_user->user_pass ) {
+				$user_password_changed = true;
+			}
 			unset( $old_user->user_pass );
 			if ( serialize( $old_user ) === serialize( $user->data ) ) {
+				if ( $user_password_changed ) {
+					do_action( 'jetpack_updated_user_password', $user );
```
Comment (enejb) [inline sync/class.jetpack-sync-module-users.php]: change to `jetpack_sync_save_user` action.

File: sync/class.jetpack-sync-module-users.php [hunk 2]
```diff
@@ -241,8 +251,9 @@ function save_user_handler( $user_id, $old_user_data = null ) {
 		 * @since 4.2.0
 		 *
 		 * @param object The WP_User object
+		 * @param array state
 		 */
-		do_action( 'jetpack_sync_save_user', $user );
+		do_action( 'jetpack_sync_save_user', $user, array( 'password_changed' => $user_password_changed ) );
```
Comment (enejb) [inline sync/class.jetpack-sync-module-users.php]: also set that the flag for user data changed.

File: sync/class.jetpack-sync-module-users.php [hunk 3]
```diff
@@ -235,19 +258,28 @@ function save_user_handler( $user_id, $old_user_data = null ) {
...
 		 *
 		 * @param object The WP_User object
+		 * @param array state - New since 5.8.0
 		 */
-		do_action( 'jetpack_sync_save_user', $user );
+		do_action( 'jetpack_sync_save_user', $user, array(
+			'password_changed' => $user_password_changed,
+			'user_data_changed' => true,
+			'role_changed' => (bool) $role_changed,
+			'previous_role' => $role_changed,
```
Comment (lezama) [inline sync/class.jetpack-sync-module-users.php]: What do you think about putting `previous_role` inside `role_changed`?

File: sync/class.jetpack-sync-module-users.php [hunk 4]
```diff
@@ -23,7 +25,8 @@ public function init_listeners( $callable ) {
 		add_action( 'add_user_to_blog', array( $this, 'save_user_handler' ) );
 		add_action( 'jetpack_sync_add_user', $callable, 10, 2 );
 		add_action( 'jetpack_sync_register_user', $callable, 10, 2 );
-		add_action( 'jetpack_sync_save_user', $callable );
+		add_action( 'jetpack_sync_save_user', $callable, 10, 2 );
+		add_action( 'jetpack_updated_user_password', $callable );
```
Comment (lezama) [inline sync/class.jetpack-sync-module-users.php]: this line can be removed now!

**Pull request merged**